hummingbot/core/api_throttler/async_throttler_base.py

```diff
@@ -0,0 +1,73 @@
+import asyncio
+import copy
+
+from abc import ABC, abstractmethod
+from decimal import Decimal
+from typing import (
+    Dict,
+    List,
+    Optional,
+)
+
+from hummingbot.client.config.global_config_map import global_config_map
+from hummingbot.core.api_throttler.async_request_context_base import AsyncRequestContextBase
+from hummingbot.core.api_throttler.data_types import (
+    RateLimit,
+    TaskLog
+)
+
+
+class AsyncThrottlerBase(ABC):
+    def __init__(self,
+                 rate_limits: List[RateLimit],
+                 retry_interval: float = 0.1,
+                 safety_margin_pct: Optional[float] = 0.05,  # An extra safety margin, in percentage.
+                 ):
+        """
+        The APIThrottlerBase is an abstract class meant to describe the functions necessary to handle the
+        throttling of API requests through the usage of asynchronous context managers.
+        :param rate_limits: List of RateLimit(s).
+        :param retry_interval: Time between every capacity check.
+        :param safety_margin: Percentage of limit to be added as a safety margin when calculating capacity to ensure calls are within the limit.
+        """
+
+        # Rate Limit Definitions
+        self._rate_limits: List[RateLimit] = copy.deepcopy(rate_limits)
+        limits_pct_conf: Optional[Decimal] = global_config_map["rate_limits_share_pct"].value
+
+        # If configured, users can define the percentage of rate limits to allocate to the throttler.
+        self.limits_pct: Optional[Decimal] = Decimal("1") if limits_pct_conf is None else limits_pct_conf // Decimal("100")
```

Comment:
I am not sure we need to consider the percentage an integer value (it should be possible for the user to specify a percentage of 33.33)

Reply:
Yea I agree with you we should not round result. Will resolve this